libmscore/style.cpp

```diff
@@ -1134,6 +1134,7 @@ static const StyleType styleTypes[] {
       { Sid::vibratoMinDistance,            "vibratoMinDistance",            Spatium(1.0)  },
       { Sid::voltaMinDistance,              "voltaMinDistance",              Spatium(1.0)  },
       { Sid::figuredBassMinDistance,        "figuredBassMinDistance",        Spatium(0.5)  },
+      { Sid::tupletMinDistance,             "tupletMinDistance",             Spatium(1.0)  },
```

Comment:
Isn't this a bit much?  It results in tuplets being further from notes than before.  I'd suggest 0.5 at most.

Reply:
Yes, vtests actually show the same, I didn't notice this initially. It is much better visible when comparing 3.1 vs. 3.2 versions but comparing to the reference files also shows this difference, see http://vtest.musescore.org/e8a16cac/vtest.html#tuplets-1. So this should probably indeed be smaller. The only thing that may need checking is an appearance of tuplets above articulations, if it doesn't look good with the lower value of this setting then some additional spacing may be required for this case.

Reply:
It looks good in the cases I tried, so I have a commit reducing it in my PR #5189

Reply:
Looking at vtests, the results of layout with this min. distance value still differ from those in 3.1 version. Reducing this value further reduces the differences but there is also another issue here that influences the result. The generic autoplacement algorithm uses the element's bounding box in order to find its proper placement related to a skyline. However this is not always correct for tuplets: tuplets with brackets have a more complex shape consisting of the bracket itself (which may have also some slope) and a number text. This makes the tuplet be sometimes unnecessarily pulled further from a staff because of the bracket edge that does not touch anything itself but makes a bounding box much higher. Calculating the distance using the element's shape instead of bbox in `autoplaceMeasureElement()` makes the differences from 3.1 version much smaller, and making this min. distance value equal to 0.3 or 0.4 makes only two vtest examples containing tuplets differ from their appearance in 3.1 version, and that difference is actually rather small (probably negligibly small). Tuplets above articulations still seem to look good with these min. distance values.

So, in conclusion, I believe the best option would be to use shapes for autoplacement for tuplets (that will work for all elements but, for performance reasons, it may be better to keep other elements using bboxes), possibly reducing the minimal autoplacement distance for tuplets to 0.4 or 0.3 of a spatium unit.

And a side question: do you think it would be useful (for anyone except me) to create something similar to [vtest.musescore.org](http://vtest.musescore.org/index.html) but with a possibility to show differences between different versions (and, possibly, vtest differences of certain pull request versus master/release versions)? I have modified vtest scripts to make it possible for me locally but this still doesn't make it easy to show the results to anyone else.